### PR TITLE
Added French File

### PR DESCRIPTION
**Pull request by aryasoni98**
> * [ ]  Base README.md
> * [x]  for-teachers.md
> * [x]  Quizzes
> * [x]  1-getting-started-lessons base README
>   
>   * [x]  intro-to-programming-languages README
>   * [x]  intro-to-programming-languages assignment
>   * [x]  github-basics README
>   * [x]  accessibility README
>   * [x]  accessibility assignment
> * [x]  2-js-basics base README
>   
>   * [x]  data-types README
>   * [x]  data-types assignment
>   * [x ]  functions-methods README
>   * [x]  functions-methods assignment
>   * []  making-decisions README
>   * [x]  making-decisions assignment
>   * [x]  arrays-loops README
>   * [x]  arrays-loops assignment
> * [x]  3-terrarium base README
>   
>   * [x]  intro-to-html README
>   * [x]  intro-to-html assignment
>   * [ ]  intro-to-css README
>   * [x]  intro-to-css assignment
>   * [ ]  intro-to-DOM-and-closures README
>   * [x]  intro-to-DOM-and-closures assignment
> * [x]  4-typing-game base README
>   
>   * [ ]  typing-game README
>   * [x]  typing-game assignment
> * [x]  5-browser-extension base README
>   
>   * [ ]  about-browsers README
>   * [ ]  about-browsers assignment
>   * [ ]  forms-browsers-local-storage README
>   * [ ]  forms-browsers-local-storage assignment
>   * [ ]  background-tasks-and-performance README
>   * [ ]  background-tasks-and-performance assignment
>   * [ ]  solution base README
>   * [ ]  start base README
> * [ ]  6-space-game base README
>   
>   * [ ]  introduction README
>   * [ ]  introduction assignment
>   * [ ]  drawing-to-canvas README
>   * [ ]  drawing-to-canvas  assignment
>   * [ ]  moving-elements-around README
>   * [ ]  moving-elements-around  assignment
>   * [ ]  collision-detection README
>   * [ ]  collision-detection assignment
>   * [ ]  keeping-score README
>   * [ ]  keeping-score assignment
>   * [ ]  end-condition README
>   * [ ]  end-condition assignment
> * [ ]  7-bank-project base README
>   
>   * [ ]  template-route README
>   * [ ]  template-route assignment
>   * [ ]  forms README
>   * [ ]  forms  assignment
>   * [ ]  data README
>   * [ ]  data assignment
>   * [ ]  state-management README
>   * [ ]  state-management  assignment
>   * [ ]  solution base README

> * [ ]  Base README.md
> * [x]  for-teachers.md
> * [x]  Quizzes
> * [x]  1-getting-started-lessons base README
>   
>   * [x]  intro-to-programming-languages README
>   * [x]  intro-to-programming-languages assignment
>   * [x]  github-basics README
>   * [x]  accessibility README
>   * [x]  accessibility assignment
> * [x]  2-js-basics base README
>   
>   * [x]  data-types README
>   * [x]  data-types assignment
>   * [ ]  functions-methods README
>   * [x]  functions-methods assignment
>   * [ ]  making-decisions README
>   * [x]  making-decisions assignment
>   * [ ]  arrays-loops README
>   * [x]  arrays-loops assignment
> * [x]  3-terrarium base README
>   
>   * [ ]  intro-to-html README
>   * [x]  intro-to-html assignment
>   * [ ]  intro-to-css README
>   * [x]  intro-to-css assignment
>   * [ ]  intro-to-DOM-and-closures README
>   * [x]  intro-to-DOM-and-closures assignment
> * [x]  4-typing-game base README
>   
>   * [ ]  typing-game README
>   * [x]  typing-game assignment
> * [x]  5-browser-extension base README
>   
>   * [ ]  about-browsers README
>   * [ ]  about-browsers assignment
>   * [ ]  forms-browsers-local-storage README
>   * [ ]  forms-browsers-local-storage assignment
>   * [ ]  background-tasks-and-performance README
>   * [ ]  background-tasks-and-performance assignment
>   * [ ]  solution base README
>   * [ ]  start base README
> * [ ]  6-space-game base README
>   
>   * [ ]  introduction README
>   * [ ]  introduction assignment
>   * [ ]  drawing-to-canvas README
>   * [ ]  drawing-to-canvas  assignment
>   * [ ]  moving-elements-around README
>   * [ ]  moving-elements-around  assignment
>   * [ ]  collision-detection README
>   * [ ]  collision-detection assignment
>   * [ ]  keeping-score README
>   * [ ]  keeping-score assignment
>   * [ ]  end-condition README
>   * [ ]  end-condition assignment
> * [ ]  7-bank-project base README
>   
>   * [ ]  template-route README
>   * [ ]  template-route assignment
>   * [ ]  forms README
>   * [ ]  forms  assignment
>   * [ ]  data README
>   * [ ]  data assignment
>   * [ ]  state-management README
>   * [ ]  state-management  assignment
>   * [ ]  solution base README

### CODE REVIEW

**Review by jlooper — APPROVED** [review]

**Comment by jlooper** [comment]
@ManuSquall would you take a look please?

**Comment by ManuSquall** [comment]
Sure, on it

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+Vous avez maintenant vu que nous avons trois façons de passer une fonction en tant que paramètre et vous vous demandez peut-être quand utiliser chacune. Si vous savez que vous utiliserez la fonction plus d'une fois, créez-la normalement. Si vous ne l'utilisez que pour un seul emplacement, il est généralement préférable d'utiliser une fonction anonyme. Que vous utilisiez ou non une grosse fonction de flèche ou la syntaxe plus traditionnelle de « fonction » dépend de vous, mais vous remarquerez que la plupart des développeurs modernes préfèrent « => ».
+
+---
+
+## 🚀 Défi
+
+Pouvez-vous articuler en une phrase la différence entre les fonctions et les méthodes ? Essaie!
+
+## Quiz post-conférence
+[Quizz post-conférence](https://happy-mud-02d95f10f.azurestaticapps.net/quiz/10)
```
Comment (ManuSquall) [inline 2-js-basics/2-functions-methods/translations/README.fr.md:188]: Have to add the localization parameter
[Quizz post-conférence](https://happy-mud-02d95f10f.azurestaticapps.net/quiz/10?loc=fr)

**Review by ManuSquall — CHANGES_REQUESTED** [review]
Thanks for the translation @aryasoni98. Please don't forget to refer the french files in the links and the quiz. I let some reviews on your first translated file.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+> **REMARQUE :** Il existe un type spécial de fonction connu sous le nom de **méthode**, que vous avez déjà utilisé ! En fait, nous l'avons vu dans notre démo ci-dessus lorsque nous avons utilisé `console.log`. Ce qui différencie une méthode d'une fonction, c'est qu'une méthode est attachée à un objet (`console` dans notre exemple), tandis qu'une fonction est flottante. Vous entendrez de nombreux développeurs utiliser ces termes de manière interchangeable.
+
+### Bonnes pratiques de la fonction
+
+Il y a une poignée de bonnes pratiques à garder à l'esprit lors de la création de fonctions
+
+- Comme toujours, utilisez des noms descriptifs pour savoir ce que la fonction fera
+- Utilisez **camelCasing** pour combiner des mots
+- Gardez vos fonctions concentrées sur une tâche spécifique
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:56]: I will review the file again and fix the bug ASAP sir .

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+## Passer des informations à une fonction
+
+Pour rendre une fonction plus réutilisable, vous souhaiterez souvent lui transmettre des informations. Si nous considérons notre exemple `displayGreeting` ci-dessus, il affichera uniquement **Hello, world!**. Pas la fonction la plus utile que l'on puisse créer. Si nous voulons le rendre un peu plus flexible, comme permettre à quelqu'un de spécifier le nom de la personne à saluer, nous pouvons ajouter un **paramètre**. Un paramètre (également parfois appelé **argument**) est une information supplémentaire envoyée à une fonction.
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:60]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+Lorsque nous voulons appeler notre fonction et passer le paramètre, nous le spécifions entre parenthèses.
+
+```javascript
+displayGreeting('Christopher');
+// affiche "Bonjour, Christophe !" quand courir
+```
+
+## Les valeurs par défaut
+
+Nous pouvons rendre notre fonction encore plus flexible en ajoutant plus de paramètres. Mais que se passe-t-il si nous ne voulons pas exiger que chaque valeur soit spécifiée ? Conformément à notre exemple de salutation, nous pourrions laisser le nom tel que requis (nous devons savoir qui nous saluons), mais nous voulons permettre à la salutation elle-même d'être personnalisée à votre guise. Si quelqu'un ne veut pas le personnaliser, nous fournissons une valeur par défaut à la place. Pour fournir une valeur par défaut à un paramètre, nous le définissons de la même manière que nous définissons une valeur pour une variable - `parameterName = 'defaultValue'`. Pour voir un exemple complet :
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:88]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+```javascript
+function displayGreeting(nom, salutation='Bonjour') {
+  console.log(`${salutation}, ${nom}`);
+}
+```
+
+Lorsque nous appelons la fonction, nous pouvons alors décider si nous voulons définir une valeur pour « salutation ».
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:96]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+setTimeout(fonction() {
+  console.log('3 secondes se sont écoulées');
+}, 3000);
+```
+
+Si vous exécutez notre nouveau code, vous remarquerez que nous obtenons les mêmes résultats. Nous avons créé une fonction, mais nous n'avons pas eu besoin de lui donner de nom !
+
+### Fonctions de la grosse flèche
+
+Un raccourci courant dans de nombreux langages de programmation (y compris JavaScript) est la possibilité d'utiliser ce qu'on appelle une fonction **flèche** ou **flèche grasse**. Il utilise un indicateur spécial de `=>`, qui ressemble à une flèche - d'où le nom ! En utilisant `=>`, nous pouvons ignorer le mot-clé `function`.
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:167]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+### Quand utiliser chaque stratégie
+
+Vous avez maintenant vu que nous avons trois façons de passer une fonction en tant que paramètre et vous vous demandez peut-être quand utiliser chacune. Si vous savez que vous utiliserez la fonction plus d'une fois, créez-la normalement. Si vous ne l'utilisez que pour un seul emplacement, il est généralement préférable d'utiliser une fonction anonyme. Que vous utilisiez ou non une grosse fonction de flèche ou la syntaxe plus traditionnelle de « fonction » dépend de vous, mais vous remarquerez que la plupart des développeurs modernes préfèrent « => ».
+
+---
+
+## 🚀 Défi
+
+Pouvez-vous articuler en une phrase la différence entre les fonctions et les méthodes ? Essaie!
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:185]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+## Quiz post-conférence
+[Quizz post-conférence](https://happy-mud-02d95f10f.azurestaticapps.net/quiz/10)
+
+## Révision et auto-apprentissage
+
+Cela vaut la peine de [lire un peu plus sur les fonctions fléchées] (https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions), car elles sont de plus en plus utilisées dans les bases de code. Entraînez-vous à écrire une fonction, puis à la réécrire avec cette syntaxe.
+
+## Mission
+
+[Amusez-vous avec les fonctions](assignment.md)
```
[the remark this answers is not in the document]
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:196]: I will review the file again and fix the bug ASAP sir.

File: 2-js-basics/2-functions-methods/translations/README.fr.md (hunk 1 above)
Comment (aryasoni98) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:188]: I will review the file again and fix the bug ASAP sir.

**Comment by aryasoni98** [comment]
@ManuSquall Sir please review my PR

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+Si je voulais créer une fonction pour afficher un message d'accueil, cela pourrait ressembler à ceci :
+
+```javascript
+fonction affichageSalutation() {
+  console.log('Bonjour tout le monde !');
+}
+```
+
+Chaque fois que nous voulons appeler (ou invoquer) notre fonction, nous utilisons le nom de la fonction suivi de `()`. Il convient de noter que notre fonction peut être définie avant ou après que nous décidions de l'appeler ; le compilateur JavaScript le trouvera pour vous.
```
[the remark this answers is not in the document]
Comment (ManuSquall) [reply 2-js-basics/2-functions-methods/translations/README.fr.md:41]: The `"le"` is still in the last sentence

File: 2-js-basics/2-functions-methods/translations/README.fr.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+> **REMARQUE :** Il existe un type spécial de fonction connu sous le nom de **méthode**, que vous avez déjà utilisé ! En fait, nous l'avons vu dans notre démo ci-dessus lorsque nous avons utilisé `console.log`. Ce qui différencie une méthode d'une fonction, c'est qu'une méthode est attachée à un objet (`console` dans notre exemple), tandis qu'une fonction est flottante. Vous entendrez de nombreux développeurs utiliser ces termes de manière interchangeable.
+
+### Bonnes pratiques de la fonction
+
+Il y a une poignée de bonnes pratiques à garder à l'esprit lors de la création de fonctions
+
+- Comme toujours, utilisez des noms descriptifs pour savoir ce que la fonction fera
+- Utilisez **boîte de chameau** pour combiner des mots
```
Comment (ManuSquall) [inline 2-js-basics/2-functions-methods/translations/README.fr.md:55]: `boîte de chameau` doesn't have sense here from my point of vue, the `camelCase` word is also used in french and `boîte de chameau` is not appropriated for the context i think

**Review by ManuSquall — CHANGES_REQUESTED** [review]
@aryasoni98 most of my reviews havn't been considered, so please recheck them, i also noticed that you changed an english basic file, maybe you should make it in another PR or change the title of this one?

Please yours link to the assignments and the quiz have also to refer the french one.

**Comment by ManuSquall** [comment]
@aryasoni98 it seems you have done some changes on the basics english files in your "added french file" PR:

image

**Comment by ManuSquall** [comment]
Is it possible for me to edit your files or i have to fork your repo?

**Comment by saan2** [comment]
plugins {
    id 'com.android.application'
    id 'kotlin-android'
}

android {
    compileSdkVersion 30
    buildToolsVersion "30.0.3"

    defaultConfig {
        applicationId "com.aespaes.sck"
        minSdkVersion 21
        targetSdkVersion 30
        versionCode 1
        versionName "1.0"

        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
    kotlinOptions {
        jvmTarget = '1.8'
    }
}

dependencies {

    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
    implementation 'androidx.core:core-ktx:1.6.0'
    implementation 'androidx.appcompat:appcompat:1.3.0'
    implementation 'com.google.android.material:material:1.4.0'
    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
    testImplementation 'junit:junit:4.+'
    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'

    implementation 'com.google.firebase:firebase-ads:20.2.0'
    implementation 'com.facebook.android:audience-network-sdk:6.+'
}

**Review by MelnikovAG — APPROVED** [review]

**Comment by jlooper** [comment]
I'm going to close this as I don't think the proper procedures are being followed per @ManuSquall 's instructions and the thread is attracting spam

**Pull request closed without merging**